src/index.ts

```diff
@@ -58,8 +58,46 @@ const formatDnsDidRecord = ({ a, v, p, type }: { [key: string]: string }) => {
   };
 };
 
-export const queryDns = async (domain: string): Promise<IDNSQueryResponse> => {
-  const { data } = await axios.get<IDNSQueryResponse>(`https://dns.google/resolve?name=${domain}&type=TXT`);
+export const queryGoogleDns = async (domain: string, headers: Record<string, string>): Promise<IDNSQueryResponse> => {
+  const { data } = await axios.get<IDNSQueryResponse>(`https://dns.google/resolve?name=${domain}&type=TXT`, {
+    headers,
+  });
+
+  return data;
+};
+
+export const queryCloudflareDns = async (
+  domain: string,
+  headers: Record<string, string>
+): Promise<IDNSQueryResponse> => {
+  const { data } = await axios.get<IDNSQueryResponse>(`https://1.1.1.1/dns-query?name=${domain}&type=TXT`, { headers });
+  return data;
+};
+
+export const queryCustomDns = async (
```

Comment:
1) I'm not sure if the `api` is the same for all DOH (like do they all have an endpoint which expect name and type?
2) how does one configure dnsprove lib to use the custom DNS

Reply:
im not sure what you mean by the second point, but maybe its resolved in the changes I made